social/utils.py

```diff
@@ -166,7 +166,22 @@ def partial_pipeline_data(backend, user=None, *args, **kwargs):
     if partial:
         idx, backend_name, xargs, xkwargs = \
             backend.strategy.partial_from_session(partial)
+
+        partial_matches_request = False
+
         if backend_name == backend.name:
+            partial_matches_request = True
+
+            req_data = backend.strategy.request_data()
+            # only need to check if current request specifies the ID_KEY
+            if backend.ID_KEY in req_data:
```

Comment:
Normally when resuming a pipeline, `request_data` will be empty. We only need to check if new data was provided.